mir_eval/sonify.py

```diff
@@ -140,6 +142,64 @@ def _fast_synthesize(frequency):
     return output
 
 
+def pitch_contour(times, frequencies, fs, function=np.sin, length=None,
+                  kind='linear'):
+    '''Sonify a pitch contour.
+
+    Parameters
+    ----------
+    times : np.ndarray
+        time indices for each frequency measurement, in seconds
+
+    frequencies : np.ndarray
+        frequency measurements, in Hz.
+        Non-positive measurements will be interpreted as un-voiced samples.
+
+    fs : int
+        desired sampling rate of the output signal
+
+    function : function
+        function to use to synthesize notes, should be 2pi-periodic
+
+    length : int
+        desired number of samples in the output signal,
+        defaults to ``max(times)*fs``
+
+    kind : str
+        Interpolation mode for the frequency estimator.
+        See: ``scipy.interpolate.interp1d`` for valid settings.
+
+    Returns
+    -------
+    output : np.ndarray
+        synthesized version of the pitch contour
+    '''
+
+    fs = float(fs)
+
+    if length is None:
+        length = int(times.max() * fs)
+
+    # Squash the negative frequencies.
+    # wave(0) = 0, so clipping here will un-voice the corresponding instants
+    frequencies = np.maximum(frequencies, 0.0)
+
+    # Build a frequency interpolator
+    f_interp = interp1d(times * fs, frequencies, kind=kind,
+                        fill_value=0.0, bounds_error=False, copy=False)
+
+    # Estimate frequency at sample points
+    f_est = f_interp(np.arange(length))
+
+    # Sonify the waveform
+    x = function(2 * np.pi * np.cumsum(f_est) / fs)
+
+    # Clamp the extrapolated values
+    x[:int(times.min() * fs)] = 0
+    x[int(times.max() * fs):] = 0
```

Comment:
> Actually it's worse than that -- the interpolator seems to be constant-padding instead of using fill_value for extrapolation.

Hmm, seems ok to me:

```Python
In [1]: import numpy as np

In [2]: import scipy.interpolate

In [3]: x = np.linspace(.2, 1.1, 10)

In [4]: y = np.random.random_integers(2, 5, 10)

In [5]: f = scipy.interpolate.interp1d(x, y, kind='slinear', fill_value=10., bounds_error=False)

In [6]: f(.15)
Out[6]: array(10.0)
```

Is that not what you'd expect?

> At any rate, the discontinuity here doesn't bother me too much.

Well, not explicitly zeroing out doesn't sound *worse*, right?  Unless there is some nasty behavior I am not seeing, it seems like interpolate will already be zeroing out things.